core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -23,377 +22,271 @@ import scala.jdk.CollectionConverters._
 import scala.math.BigDecimal.RoundingMode
 
 object SpendReportingService {
-  def constructor(dataSource: SlickDataSource,
-                  bigQueryService: cats.effect.Resource[IO, GoogleBigQueryService[IO]],
-                  samDAO: SamDAO,
-                  spendReportingServiceConfig: SpendReportingServiceConfig
+  def constructor(
+    dataSource: SlickDataSource,
+    bigQueryService: cats.effect.Resource[IO, GoogleBigQueryService[IO]],
+    samDAO: SamDAO,
+    spendReportingServiceConfig: SpendReportingServiceConfig
   )(ctx: RawlsRequestContext)(implicit executionContext: ExecutionContext): SpendReportingService =
     new SpendReportingService(ctx, dataSource, bigQueryService, samDAO, spendReportingServiceConfig)
-}
 
-class SpendReportingService(ctx: RawlsRequestContext,
-                            dataSource: SlickDataSource,
-                            bigQueryService: cats.effect.Resource[IO, GoogleBigQueryService[IO]],
-                            samDAO: SamDAO,
-                            spendReportingServiceConfig: SpendReportingServiceConfig
-)(implicit val executionContext: ExecutionContext)
-    extends LazyLogging {
-  private def requireProjectAction[T](projectName: RawlsBillingProjectName, action: SamResourceAction)(
-    op: => Future[T]
-  ): Future[T] =
-    samDAO.userHasAction(SamResourceTypeNames.billingProject, projectName.value, action, ctx.userInfo).flatMap {
-      case true => op
-      case false =>
-        Future.failed(
-          new RawlsExceptionWithErrorReport(
-            errorReport = ErrorReport(
-              StatusCodes.Forbidden,
-              s"${ctx.userInfo.userEmail.value} cannot perform ${action.value} on project ${projectName.value}"
-            )
-          )
+  def extractSpendReportingResults(
```

Comment:
I moved this to the companion object to:
1. Make it more clear that it was a pure function that didn't rely on any class resources
2. Simplify unit testing a bit by removing the need to instantiate the service with all the mocked resources